modules/p2/ting_openlist/include/ting_openlist_analysis.inc

```diff
@@ -0,0 +1,150 @@
+ */
+function ting_openlist_local_analysis_data_providers_id() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_uid', NULL, 'IS NOT');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Get providers which is synced.
+ *
+ * @return int
+ *   Number of synced users.
+ */
+function ting_openlist_local_analysis_data_providers_openlist_is_sync() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_modified', 0, '>');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Count the types of lists.
+ *
+ * @return array
+ *   An associative array, where the key is the list_type_X (X being number of
+ *   lists) and the key is number of occurences.
+ */
+function ting_openlist_local_analysis_data_list_counts() {
+  $sub_query = db_select('field_data_field_list_type', 'flt');
+  $sub_query->join('eck_ding_type', 'dt', 'dt.id = flt.entity_id');
+  $sub_query->addField('flt', 'field_list_type_value', 'type');
+  $sub_query->addExpression('COUNT(flt.entity_id)', 'cnt');
+  $sub_query
+    ->condition('flt.field_list_type_value', 'user_list', '!=')
+    ->groupBy('dt.uid')
+    ->groupBy('flt.field_list_type_value')
+    ->orderBy('cnt', 'DESC');
+  $query = db_select($sub_query, 's');
+  $query->addExpression("CONCAT(s.type, '__', s.cnt)", 'type');
+  $query->addExpression('COUNT(s.type)', 'cnt');
+  $query
+    ->groupBy('s.type')
+    ->groupBy('s.cnt')
+    ->orderBy('s.type')
+    ->orderBy('s.cnt');
+  return $query->execute()->fetchAllKeyed();
+}
+
+/**
+ * The top 10 duplicate lists.
+ *
+ * @return array
+ *   Keyed array, where they key is the openlist_uid and value the number of
+ *   list duplicates.
+ */
+function ting_openlist_local_analysis_data_top10_list() {
+  $sub_query = db_select('field_data_field_list_type', 'flt');
+  $sub_query->join('eck_ding_type', 'dt', 'dt.id = flt.entity_id');
+  $sub_query->addField('dt', 'uid');
+  $sub_query->addExpression('COUNT(flt.entity_id)', 'cnt');
+  $sub_query
+    ->condition('flt.field_list_type_value', 'user_list', '!=')
+    ->groupBy('dt.uid')
+    ->groupBy('flt.field_list_type_value')
+    ->orderBy('cnt', 'DESC')
+    ->havingCondition('cnt', 1, '>');
+  $query = db_select($sub_query, 's');
+  $query->join('users', 'u', 'u.uid = s.uid');
+  $query->addField('u', 'openlist_uid');
+  $query->addField('s', 'cnt');
+  $query
+    ->groupBy('s.uid')
+    ->groupBy('s.cnt')
+    ->orderBy('cnt', 'DESC')
+    ->range(0, 10);
+  return $query->execute()->fetchAllKeyed();
+}
+
+/**
+ * Get lists with duplicate elements.
+ *
+ * @return array
+ *   An associative array, where the key tells how many duplicates there is, and
+ *   the value is the number of occurences.
```

Comment:
Again I might be misunderstanding something. The DocBlock says that the function returns "lists with duplicate elements".

As I read it does not return any actual lists but rather the number of times a list duplicates occur. Example: If 10 lists have been duplicated 3 times and, 7 lists have been duplicated 2 times then the method would return

```php
[
  3 => 10,
  2 => 7,
]
```